consent/api/v1/views.py

```diff
@@ -149,14 +169,21 @@ def get(self, request):
 
         """
         try:
+            # Ensure that we got the required explicit parameters in the API
             username, course_id, enterprise_customer_uuid = self.get_required_query_params(request)
         except ConsentAPIRequestError as invalid_request:
             return Response({'error': str(invalid_request)}, status=HTTP_400_BAD_REQUEST)
 
         user_id = get_user_id(username)
         exists = consent_exists(user_id, course_id, enterprise_customer_uuid)
         provided = consent_provided(user_id, course_id, enterprise_customer_uuid)
-        required = consent_required(user_id, course_id, enterprise_customer_uuid)
+        enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+        if enterprise_customer and not enterprise_customer.catalog_contains_course_run(request.user, course_id):
+            # Only mark consent as required if it's in the Enterprise-linked
+            # course catalog; not otherwise.
+            required = False
+        else:
+            required = consent_required(user_id, course_id, enterprise_customer_uuid)
```

Comment:
@douglashall @haikuginger I'm currently doing this in my most recent commit for #168 -- see https://github.com/edx/edx-enterprise/pull/168/commits/a754904e0b7d9a805f4b383a544ad49933105ff4#diff-1acf575642d97e79b9692af567dbe92dL162 and below, if you'd like to re-use the approach.